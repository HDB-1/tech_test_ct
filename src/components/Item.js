import React from 'react'

export default function Item(props) {
    const itemComp = {
        border: '2px solid black',
        margin: '5px'
    }
    return (
        <div style={itemComp}>
            <p>Name: {props.allInfo.name}</p>
            <p>Monthly Expected Cost: £{props.allInfo.expectedMonthlySpend}</p>
            <p>Expected Annual Saving: £{props.allInfo.expectedAnnualSavings}</p>
            {/* {props. ? "show fixed ones" : "no state for fixed yet"} */}
        </div>
    )
}
