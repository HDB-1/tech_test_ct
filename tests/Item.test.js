import React from 'react';
import { shallow } from 'enzyme';
import Item from '../src/components/Item';

describe("shallow Item", () => {
    let wrapper
    let showFixedMock = jest.fn() 
    let showVariableMock = jest.fn()
    let all = {
        fixed: false,
        variable: false
    }
    let energyData = [{
        id: "tariffSelectionE716598",
        supplyDetailsId: "1795",
        canApply: true,
        name: "Covalent Capped With Fee",
        logoUri: "https://rest.staging.energyhelpline.com/Uploads/SupplierLogos/Woodland_TRANS.gif?t=QtT9F_RfNGiloVcwLpCkGELowyI",
        expectedAnnualSavings: 112,
        expectedAnnualSpend: "469.67",
        expectedMonthlySpend: "39.14",
        messages: [
        "Early exit fee: Â£30.00"
        ],
        tags: [ ],
        tariffType: "unknown",
        paymentMethod: "Monthly Direct Debit",
        supplierName: "Covalent Energy",
        supplierId: "136",
        isBigSix: false,
        hasPaperBilling: true,
        hasExitFees: true
        },
        {
        id: "tariffSelectionE716668",
        supplyDetailsId: "1796",
        canApply: true,
        name: "Covalent Fixed Six Months",
        logoUri: "https://rest.staging.energyhelpline.com/Uploads/SupplierLogos/Woodland_TRANS.gif?t=QtT9F_RfNGiloVcwLpCkGELowyI",
        expectedAnnualSavings: 112,
        expectedAnnualSpend: "469.67",
        expectedMonthlySpend: "39.14",
        messages: [
        "No early exit fees",
        "Fixed until 26 February 2017"
        ],
        tags: [ ],
        tariffType: "fixed",
        paymentMethod: "Monthly Direct Debit",
        supplierName: "Covalent Energy",
        supplierId: "136",
        isBigSix: false,
        hasPaperBilling: true,
        hasExitFees: false
        }]

    beforeEach(() => {
        wrapper = shallow(<Item energyData={energyData} showFixed={showFixedMock} showVariable={showVariableMock} all={all}/>)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
})