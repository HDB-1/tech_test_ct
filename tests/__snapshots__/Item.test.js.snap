// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Item should render correctly 1`] = `
<div
  style={
    Object {
      "border": "2px solid black",
      "margin": "5px",
    }
  }
>
  <p>
    Name: 
  </p>
  <p>
    Monthly Expected Cost: £
  </p>
  <p>
    Expected Annual Saving: £
  </p>
</div>
`;
